contracts/AINFT721.sol

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Enumerable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/interfaces/IERC721.sol";
+
+import "./interfaces/IERC4906.sol";
+import "./interfaces/IAINFT.sol";
+
+/**
+ *@dev AINFT721 contract
+ */
+contract AINFT721 is
+    ERC721Enumerable,
+    Pausable,
+    AccessControl,
+    ERC721Burnable,
+    IERC4906,
+    IAINFT
+{
+    using Counters for Counters.Counter;
+    using Strings for uint256;
+
+    struct MetadataContainer {
+        address updater;
+        string metadataURI;
+    }
+    bool public immutable IS_CLONED;
+    IERC721 public immutable ORIGIN_NFT;
+    address public PAYMENT_PLUGIN;
+    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+
+    Counters.Counter private _tokenIdCounter;
```

Comment:
I think it can be counted without Counter stuff. What about simply counting it by recording lastTokenId if it increases only one when minted?

Reply:
Why we need to count `_tokenIdCounter`? All `tokenId`'s `owner`/base`Metadata` are the same as clonedNFT's `tokenID`, `owner`, `metadata`. isn't it?

Reply:
this was not tracked with `mintFromOriginInstance` and `mintFromOriginInstanceOnBehalf` at all. remove it or track these.

Reply:
got it. - resolved.